C++20 related fixes

ktf opened the pull request:
- Drop unneeded pragma
- Use std compliant __VA_OPT__ rather than ##__VA_ARGS__ extension.

alibuild commented:
Error while checking build/O2/fullCI for 79a2e5e43808ad10ba6144def34f8f0b567328ed at 2025-01-13 22:26:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/O2Physics-latest/log
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/13860/79a2e5e43808ad10ba6144def34f8f0b567328ed/build_O2_fullCI/pretty.html).

ktf committed:
C++20 related fixes

- Drop unneeded pragma
- Use std compliant __VA_OPT__ rather than ##__VA_ARGS__ extension.

alibuild commented:
Error while checking build/O2/fullCI for d5d6af829d4a1ab94b2a057245dda693453b41fe at 2025-01-14 03:07:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/13860/d5d6af829d4a1ab94b2a057245dda693453b41fe/build_O2_fullCI/pretty.html).
